src/pps/filter.go

```diff
@@ -0,0 +1,22 @@
+package pps
+
+import (
+	context "context"
+)
+
+func (r *ListDatumRequest_Filter) Allow(ctx context.Context, item *DatumInfo) bool {
```

Comment:
Why does this take a context?

Reply:
The original reason was two-fold. One is practical: to enable more advanced filters that had to do more work to handle timeouts. The other is to be able to pull in request-scoped values when logging, for example the method name being called.

I think that in general every exported function should take an initial context argument, and omit one only by exception.

Reply:
I don't think it would make sense to have more advanced logic in a function that is being applied to every datum in a stream. I think of this as more of a utility function and more advanced logic (i.e. looking up indexes) would be better handled at a layer above. Also, I generally don't like adding complexity (context, more complicated match check) to code that could be easily changed later to support it. It can make sense when dealing with external APIs or code that has a strong probability of needing it, but I generally err on the side of simplicity.